[2024/07/11] 보드 기능

2minus opened the pull request:
- 보드 다건 조회 구현
     - 전체 보드 조회
     - 사용자 기준 전체 보드 조회

- 보드 단건 조회 구현
  - 하위 엔티티 조회 확인 필요 섹션>카드상세 까지

- 보드 생성 구현
    - 보드를 생성한 사용자에게 OWNER 권한 부여하도록 설정 (권한 설명은 후술)

- 보드 수정, 삭제 구현
    -  OWNER 권한 확인 후 기능 수행

- 보드에 사용자 추가 (초대) 구현
    - 현재는 OWNER 권한만 가능 하도록 구현

> BoardInvitation.java 에 사용자-보드 간 권한 생성
> - OWNER : 보드 생성 사용자 보드의 수정, 삭제 및 하위 게시물에 대한 수정, 삭제 권한
> - EDITOR : 섹션 이하의 게시물에 대한 수정, 삭제 권한
> - NORMAL : 자신이 작성한 섹션 이하의 게시물에 대한 수정, 삭제 권한

2minus committed:
[#3] feat 보드 기능 구현

- 보드 단건 조회 구현
     -> 🛠️섹션과 카드 생성 후 잘 반환되는지 확인 필요
- 보드 다건 조회 구현
     -> 전체 조회 / 사용자 보드 조회
- 보드 생성 구현
- 보드 수정 구현
- 보드 삭제 구현
- 보드에 사용자 추가 (초대) 구현

2minus committed:
[#3] test 보드 기능 테스트 http 파일 생성

2minus committed:
Merge remote-tracking branch 'origin/dev' into feat/board

# Conflicts:
#	src/main/java/com/gamja/trello/controller/SectionController.java

2minus committed:
[#3] fix : 하위 Entity 조희를 위한 보드 단건 조회 관련 수정

- Card.java 동기화 (필드명)
- CaerResponseDto.java 임시 추가

hyun1202 commented on src/main/java/com/gamja/trello/service/BoardService.java:
```diff
@@ -0,0 +1,194 @@
+            BoardInvitation invitation = BoardInvitation.builder()
+                    .board(board)
+                    .user(userRepository.findUserByEmail(request.getEmail()))
+                    .build();
```
1. 보드 초대 시 이미 초대된 사람들에 대한 예외처리가 없어보입니다.
2. role이 not null인데 role을 넣어주고 있지 않습니다.
확인 부탁드립니다

2minus committed:
[#3] fix : responseDto에 Entity Id 추가

2minus committed:
[#3] fix : for each 문 stream 으로 수정

2minus committed:
[#3] fix :

- for each 문 stream 으로 수정
- RefeshToken-User Cascade
- InviteRequestDto role 추가
- BoardService.inviteBoard 보드에 포함된 사용자는 초대 요청 못보내도록 제약조건 추가

2minus committed:
Merge remote-tracking branch 'origin/dev' into feat/board

# Conflicts:
#	src/main/java/com/gamja/trello/entity/Card.java
